### PR TITLE
Implement detection for "MultiDex Packer"

### PR DESCRIPTION
**Pull request by cryptax**
This packer is found in :
- `49d167f8f7427f0340297ae1c89ce4a216a8e64c55294f8e422f1f972732bae7`
- `5b9049c392eaf83b12b98419f14ece1b00042592b003a17e4e6f0fb466281368` (see [this blog post for info on this packer](https://cryptax.medium.com/multidex-trick-to-unpack-android-bianlian-ed52eb791e56)).

Beware, both samples are malicious.
I don't know what is the official name of this packer.

The packing mechanism consists in re-implementing Multidex support, while obfuscating important strings (e.g `multidex.version`) and decrypting the payload. The code fetches the payload from the assets and decrypts it. From the decrypted version, it creates a Zip file containing the DEX and loads that.

The detection rule matches (1) the de-obfuscation of some of the typical Multidex strings, and (2) the decryption of the asset.

It detects the 2 APKs with name "MultidexPacker":

```
apkid ~/samples/multidex/*.apk
[+] APKiD 2.1.3 :: from RedNaga :: rednaga.io
[*] /home/axelle/samples/multidex/08362_Video_Oynat.apk!classes.dex
 |-> anti_vm : Build.BOARD check, Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, device ID check, emulator file check, network operator name check, possible Build.SERIAL check, possible VM check, ro.kernel.qemu check, ro.product.device check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> packer : MultidexPacker
[*] /home/axelle/samples/multidex/49d167f8f7427f0340297ae1c89ce4a216a8e64c55294f8e422f1f972732bae7.apk!classes.dex
 |-> anti_disassembly : illegal class name
 |-> anti_vm : Build.BOARD check, Build.FINGERPRINT check, Build.HARDWARE check, Build.MANUFACTURER check, Build.MODEL check, Build.PRODUCT check, Build.TAGS check, SIM operator check, device ID check, emulator file check, network operator name check, possible Build.SERIAL check, possible VM check, possible ro.secure check, ro.kernel.qemu check, ro.product.device check, subscriber ID check
 |-> compiler : dexlib 2.x
 |-> packer : MultidexPacker
```

I have also tested the rule against a bunch of other APKs that do not have this packer, and did not get any False Positive. This is not bullet proof, but it looks like it is working :)

Feel free to *rename* the packer is you know what is its official name.

### CODE REVIEW

File: apkid/rules/dex/packers.yara [hunk 1]
```diff
@@ -444,3 +444,47 @@ rule jsonpacker : packer
      and ($algo or $algo2 or $algo3)
      and $dexclass
 }
+
+rule multidexpacker : packer
+{
+  meta:
```
Comment (enovella) [inline apkid/rules/dex/packers.yara:450]: Would you add the blog post in the `url` metadata field?  https://cryptax.medium.com/multidex-trick-to-unpack-android-bianlian-ed52eb791e56

**Comment by cryptax** [comment]
For the samples, they are on Koodous

- https://koodous.com/apks/5b9049c392eaf83b12b98419f14ece1b00042592b003a17e4e6f0fb466281368/general-information
- https://koodous.com/apks/49d167f8f7427f0340297ae1c89ce4a216a8e64c55294f8e422f1f972732bae7/general-information

**Comment by enovella** [comment]
LGTM @cryptax ! Many thanks...

**Comment by cryptax** [comment]
ok so I'll re-submit with the fixes.

**Pull request closed without merging**